File: 0152-maximum-product-subarray/0152-maximum-product-subarray.js
var maxProduct = function(nums) {
    let prefixSum = 1;
    let suffixSum = 1;
    let maximumProduct = -Infinity;
    const n = nums.length;
    for(let i=0;i<n;i++){
        prefixSum = prefixSum*nums[i];
        suffixSum = suffixSum*nums[n-i-1];
        maximumProduct = Math.max(maximumProduct, Math.max(prefixSum,suffixSum));
        if(prefixSum === 0) {
            prefixSum = 1;

        }
        if(suffixSum === 0) {
            suffixSum = 1;
        }
    }
    return maximumProduct;
};

File: 0152-maximum-product-subarray/0152-maximum-product-subarray.java
class Solution {
    public int maxProduct(int[] nums) {
        int n=nums.length;
        double ans=Integer.MIN_VALUE;
        double pref=1,suff=1;
        
        for(int i=0;i<n;i++)
        {
            if(pref==0)
                pref=1;
            if(suff==0)
                suff=1;
            
            pref*=nums[i];
            suff*=nums[n-i-1];
            ans=Math.max(ans,Math.max(pref,suff));
        }
        return (int)ans;
        
    }
}